cli/cliui/agent.go

```diff
@@ -389,15 +391,32 @@ func ConnDiagnostics(w io.Writer, d ConnDiags) {
 		return
 	}
 
-	if d.ConnInfo != nil && d.ConnInfo.DERPMap != nil && !d.ConnInfo.DERPMap.HasSTUN() {
-		_, _ = fmt.Fprint(w, "✘ The DERP map is not configured to use STUN, which will prevent direct connections from starting outside of local networks\n")
+	if d.ConnInfo != nil && d.ConnInfo.DERPMap != nil {
```

Comment:
This seems needlessly defensive.  If we can't get the AgentConnectionInfo from the Coder server something is very seriously wrong and p2p is the least of your worries.  Just make ConnInfo a non-pointer type and bail out way before we get here if you can't get it.

Similarly, if the DERPMap is missing, you have big problems.